elytron/ELY-2078-encryption-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,214 @@
+= [ELY-2078] Add encryption and integrity support to FilesystemSecurityRealm
+:author:            Ashpan Raskar
+:email:             araskar@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This objective of this proposal is to securely encrypt the identities and credentials in the `FileSystemSecurityRealm` stored on the local filesystem. In the past, the default for storing identities was via property files, but in future versions of WildFly, the plan is to use a FileSystemSecurityRealm as the default out of the box configuration instead. The FileSystemSecurityRealm currently uses Base64 to encode clear passswords, hashed passwords, and attributes, but we want to add the ability to encrypt the credentials and attributes for an identity. In order to do this, we plan to add SecretKey support, which is a symmetric key to encrypt and decrypt data.
+
+The objective is to encrypt the password, and attributes (eg: roles) with a SecretKey, the result of that will be a Base64 encoded string. The encrypted fields will be stored in the identity xml file.
+
+To store the SecretKey, a credential store will be created and the FileSystem realm will utilize the credential store to access the SecretKey to encrypt the FileSystem realm.
+
+For the default out of box configuration for WildFly, the goal is for the FileSystemSecurityRealm to make use of ScramDigest with AES encryption enabled.
```

Comment:
Just a small comment but would be good to update the overview to also mention integrity.